aws/Vault-Single-Deployment.yaml

```diff
@@ -172,6 +172,129 @@ Resources:
     DependsOn:
       - LambdaDeployRole
       - VaultMachine
+  LambdaRemovePermissionsRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Policies:
+        - PolicyName: CloudWatch
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogGroup
+                  - logs:CreateLogStream
+                  - logs:DescribeLogGroups
+                  - logs:DescribeLogStreams
+                  - logs:PutLogEvents
+                Resource:
+                  - '*'
+        - PolicyName: KMSPolicy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - kms:DescribeKey
+                Resource:
+                  - '*'
```

Comment:
is this open because we don't know the name of the key during vault deployment?

Reply:
right, in the creation time of the role we don't know the kms id and in addition the kms is created outside of the stack so we can't refer to it.